app/models/vacancy.rb

```diff
@@ -68,6 +69,26 @@ class Vacancy < ApplicationRecord
       self.status
     end
 
+    attribute :job_role_0 do
```

Comment:
You can do all of these in a single loop:

```ruby 
self.job_roles.each_with_index do |role, index|
  attribute "job_role_#{index}" { role }
end 
```

Reply:
Unfortunately, you can't. `self` here refers to `AlgoliaSearch::IndexSettings`.

Reply:
but I found a way